ioctl/cmd/ws/wsproject.go

```diff
@@ -178,3 +205,49 @@ func getEventInputsByName(logs []*iotextypes.Log, eventName string) (map[string]
 
 	return inputs, nil
 }
+
+// upload content to endpoint, returns fetch url, content hash and error
+func upload(endpoint string, filename, hashstr string) (url string, hash256b *hash.Hash256, err error) {
+	// read file content
+	var content []byte
+	content, err = os.ReadFile(filename)
+	if err != nil {
+		err = errors.Wrap(err, errProjectConfigReadFailed.Error())
+		return
+	}
+
+	// calculate and validate hash
+	hash256b = new(hash.Hash256)
+	*hash256b = hash.Hash256b(content)
+	if hashstr != "" {
+		var hashInput hash.Hash256
+		hashInput, err = hash.HexStringToHash256(hashstr)
+		if err != nil {
+			return
+		}
+		if hashInput != *hash256b {
+			err = errProjectConfigHashUnmatched
+			return
+		}
+	}
+
+	// upload content to ipfs endpoint
+	var (
+		sh  = shell.NewShell(endpoint)
+		cid string
+	)
+	cid, err = sh.Add(bytes.NewReader(content))
+	if err != nil {
+		return "", nil, errors.Wrap(err, errUploadProjectConfigFailed.Error())
+	}
+
+	err = sh.Pin(cid)
+	if err != nil {
+		return "", nil, errors.Wrap(err, errUploadProjectConfigFailed.Error())
+	}
+
+	// generate fetch url
+	url = fmt.Sprintf("http://%s/ipfs/%s", wsProjectIPFSGatewayEndpoint, cid)
```

Comment:
Should be determined by the gateway endpoint for whether to use HTTP or HTTPS?

Reply:
yes, i will make the gateway endpoint with scheme